metadata/program/src/state/metadata.rs

```diff
@@ -0,0 +1,110 @@
+use super::UNINITIALIZED_VERSION;
+use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
+use solana_program::{
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack, Sealed},
+    pubkey::Pubkey,
+};
+
+/// STRUCT VERSION
+pub const METADATA_VERSION: u8 = 1;
+
+/// max name length
+pub const NAME_LENGTH: usize = 32;
+
+/// max symbol length
+pub const SYMBOL_LENGTH: usize = 10;
+
+/// max uri length
+pub const URI_LENGTH: usize = 200;
+
+/// Metadata
+#[derive(Clone)]
+pub struct Metadata {
+    ///version
+    pub version: u8,
```

Comment:
The version field doesn't seem helpful.  If we ever need to make an incompatible change then we'd use a new program address